File: app/src/main/java/me/dara/memoapp/ui/auth/AuthViewModel.java
package me.dara.memoapp.ui.auth;

import android.app.Application;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;

import me.dara.memoapp.AppModule;
import me.dara.memoapp.MemoApp;
import me.dara.memoapp.network.model.ApiResponse;
import me.dara.memoapp.network.model.User;

/**
 * @author sardor
 */

public final class AuthViewModel extends AndroidViewModel {
  private final AppModule module;

  public final LiveData<ApiResponse> signUp(User user){
    return this.module.memoService.signUp(user);
  }

  public final LiveData<ApiResponse> signIn( String email,  String password) {
    return this.module.memoService.signIn(email, password);
  }

  public AuthViewModel( Application app) {
    super(app);
    this.module = ((MemoApp) this.getApplication()).appModule;
  }
}

File: app/src/main/java/me/dara/memoapp/ui/auth/AuthViewModel.kt
package me.dara.memoapp.ui.auth

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import me.dara.memoapp.MemoApp

/**
 * @author sardor
 */
class AuthViewModel(app: Application) : AndroidViewModel(app) {

  val module = getApplication<MemoApp>().appModule

  fun signUp(email: String, password: String) = module.webService.signUp(email, password)

  fun signIn(email: String, password: String) = module.webService.signUp(email, password)

}